sound/soc/sof/sof-pci-dev.c

```diff
@@ -222,19 +222,25 @@ static int sof_pci_probe(struct pci_dev *pci,
 #else
 	/* find machine */
 	mach = snd_soc_acpi_find_machine(desc->machines);
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 	if (!mach) {
+		dev_warn(dev, "No matching ASoC machine driver found - falling back to HDA codec\n");
+		mach = snd_soc_acpi_intel_hda_machines;
+		mach->sof_fw_filename = desc->nocodec_fw_filename;
+
+		/*
+		 * TODO: we need to find a way to check if codecs are actually
+		 * present
+		 */
+	}
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_NOCODEC)
+	if (!mach) {
 		/* fallback to nocodec mode */
```

Comment:
We may need to make them mutually exclusive for CONFIG_SND_SOC_SOF_HDA and CONFIG_SND_SOC_SOF_NOCODEC then, or using something like #if and #else if?

Reply:
There is a mistake here. The idea was to deal with HDaudio and then nocodec. there's a missing endif... Not sure how it ever compiled

Reply:
fixed now